### PR TITLE
PMM-2717 Removed metrics "mongodb_mongos_sharding_balancer_lock_*"

### PR DESCRIPTION
**Pull request by DexterHD**
Those metrics have removed due to design changes in MongoDB 3.6+. Starting in version 3.6, the balancer no longer takes a “lock” so those metrics have no sense. For more information see: "https://docs.mongodb.com/v3.6/reference/config-database/#config.locks". For previous versions of MongoDB, those metrics will be working as before.

### CODE REVIEW

File: shared/utils.go [hunk 1]
```diff
@@ -48,3 +52,28 @@ func GetCallerLocation() string {
...
+		return false
+	}
+
+	v3dot6, err := semver.NewVersion(version)
+	if err != nil {
+		log.Errorf("couldn't parse version '%s', reason: %v", version, err)
+		return false
+	}
+
+	if v.LessThan(v3dot6) {
```
Comment (BupycHuk) [inline shared/utils.go]: return v.LessThan(v3dot6)

File: shared/utils.go [hunk 2]
```diff
@@ -48,3 +52,28 @@ func GetCallerLocation() string {
...
+		return false
+	}
+
+	v, err := semver.NewVersion(serverVersion)
+	if err != nil {
+		log.Errorf("couldn't parse mongo server version '%s', reason: %v", serverVersion, err)
+		return false
+	}
+
+	v3dot6, err := semver.NewVersion(version)
```
Comment (BupycHuk) [inline shared/utils.go]: is it always 3.6 version?

File: collector/mongos/sharding_status.go [hunk 3]
```diff
@@ -221,20 +226,28 @@ func (status *ShardingStats) Describe(ch chan<- *prometheus.Desc) {
 	balancerChunksBalanced.Describe(ch)
 	mongosUpSecs.Describe(ch)
 	mongosPing.Describe(ch)
-	mongosBalancerLockState.Describe(ch)
-	mongosBalancerLockTimestamp.Describe(ch)
+
+	if shared.MongoServerVersionLessThan("3.6", status.Client) {
```
Comment (BupycHuk) [inline collector/mongos/sharding_status.go]: why do we use status variable name for stats?

File: shared/utils.go (hunk 1 above)
Comment (DexterHD) [reply shared/utils.go]: Thx.

File: shared/utils.go (hunk 2 above)
Comment (DexterHD) [reply shared/utils.go]: No. Fixed.

File: collector/mongos/sharding_status.go (hunk 3 above)
Comment (DexterHD) [reply collector/mongos/sharding_status.go]: Historical thing. :) I've changed it to `s`.

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/mongodb_exporter?pullRequest=161) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/mongodb_exporter?pullRequest=161) before we can accept your contribution.<br/><hr/>**Anton Kucherov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/mongodb_exporter?pullRequest=161) it.</sub>

**Comment by percona-csalguero** [comment]
Tested with MongoDB cluster having shards, replicasets and config servers.
Ok
MongoDB 3.4
```
bin/mongodb_exporter --mongodb.uri=mongodb://admin:admin123456@127.0.0.1/admin --log.level=debug
INFO[0000] Starting mongodb_exporter (version=, branch=, revision=)  source="mongodb_exporter.go:80"
INFO[0000] Build context (go=go1.13.8, user=, date=19691231-21:00:00)  source="mongodb_exporter.go:81"
DEBU[0000] Connected to: mongodb://****:****@127.0.0.1/admin (node type: mongod, server version: 3.4.22-2.20)  source="mongodb_collector.go:208"
DEBU[0000] Collecting Server Status                      source="mongodb_collector.go:261"
INFO[0000] Starting HTTP server for http://:9216/metrics ...  source="server.go:140"
```

MongoDB 3.6
```
bin/mongodb_exporter --mongodb.uri=mongodb://admin:admin123456@127.0.0.1/admin --log.level=debug
INFO[0000] Starting mongodb_exporter (version=, branch=, revision=)  source="mongodb_exporter.go:80"
INFO[0000] Build context (go=go1.13.8, user=, date=19691231-21:00:00)  source="mongodb_exporter.go:81"
DEBU[0000] Connected to: mongodb://****:****@127.0.0.1/admin (node type: mongod, server version: 3.6.13-3.3)  source="mongodb_collector.go:208"
DEBU[0000] Collecting Server Status                      source="mongodb_collector.go:261"
INFO[0000] Starting HTTP server for http://:9216/metrics ...  source="server.go:140"
```

**Comment by artemgavrilov** [comment]
@DexterHD could you please sign CLA?

**Comment by AlekSi** [comment]
@DexterHD wrote that code when Percona employed him. There is no need to sign a CLA for him.

**Pull request merged**